server/src/main/java/com/example/tyfserver/payment/domain/Payment.java

```diff
@@ -0,0 +1,74 @@
+package com.example.tyfserver.payment.domain;
+
+import com.example.tyfserver.common.domain.BaseTimeEntity;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+
+import javax.persistence.*;
+
+@Entity
+@Getter
+@NoArgsConstructor(access = AccessLevel.PROTECTED)
+public class Payment extends BaseTimeEntity {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @Column(nullable = false)
+    private Long amount;
+
+    private String email;
+
+    @Column(nullable = false)
+    private String pageName;
+
+    @Enumerated(value = EnumType.STRING)
+    private PaymentStatus status = PaymentStatus.PENDING;
+
+    private String impUid;
+
+    public Payment(Long id, Long amount, String email, String pageName) {
+        this.id = id;
+        this.amount = amount;
+        this.email = email;
+        this.pageName = pageName;
+    }
+
+    public Payment(Long amount, String email, String pageName) {
+        this(null, amount, email, pageName);
+    }
+
+    public void complete(PaymentInfo paymentInfo) {
+        validatePaymentComplete(paymentInfo);
+        this.impUid = paymentInfo.getServiceId();
+        this.status = PaymentStatus.PAID;
+    }
+
+    public void updateStatus(PaymentStatus paymentStatus) {
+        this.status = paymentStatus;
+    }
+
+    private void validatePaymentComplete(PaymentInfo paymentInfo) {
+        if (!PaymentStatus.isPaid(paymentInfo.getStatus())) {
+            updateStatus(paymentInfo.getStatus());
+            throw new RuntimeException();
+        }
+
+        if (!id.equals(paymentInfo.getId())) {
+            updateStatus(PaymentStatus.INVALID);
+            throw new RuntimeException();
+        }
+
+        if (!amount.equals(paymentInfo.getAmount())) {
+            updateStatus(PaymentStatus.INVALID);
+            throw new RuntimeException();
+        }
+
+        if (!pageName.equals(paymentInfo.getPageName())) {
+            updateStatus(PaymentStatus.INVALID);
+            throw new RuntimeException();
+        }
```

Comment:
이 로직은 내가 잘 못 이해하고 있는걸까?? PAID가 아닌 경우에 다른 모든 정보가 일치하면, 상태를 paymentInfo의 status로 업데이트 해야하는거 아닌가?? 🤔
```suggestion
        if (!PaymentStatus.isPaid(paymentInfo.getStatus())) {
            throw new RuntimeException();
        }

        if (!id.equals(paymentInfo.getId())) {
            updateStatus(PaymentStatus.INVALID);
            throw new RuntimeException();
        }

        if (!amount.equals(paymentInfo.getAmount())) {
            updateStatus(PaymentStatus.INVALID);
            throw new RuntimeException();
        }

        if (!pageName.equals(paymentInfo.getPageName())) {
            updateStatus(PaymentStatus.INVALID);
            throw new RuntimeException();
        }
        updateStatus(paymentInfo.getStatus());
```

Reply:
그런가 Paid가 아닌 경우에는 전달받은 정보로 상태를 업데이트하고 결제 안하기로 한거 같아서